### PR TITLE
Improve threading for crashlytics network state tracking example

### PR DESCRIPTION
**Pull request by jakeouellette**
(Address Feedback from Marc Richards)

- Add handling for deregistering the network state handler
- Give an example of deregistering within an activity.
(one might consider registering within the application level of the class, as well)

### CODE REVIEW

**Review by samtstern — APPROVED** [review]

**Comment by samtstern** [comment]
LGTM once you get the CI to pass!

File: crash/app/src/main/java/com/google/samples/quickstart/crash/kotlin/MainActivity.kt [hunk 1]
```diff
@@ -22,13 +22,14 @@ import com.google.samples.quickstart.crash.databinding.ActivityMainBinding
...
     private lateinit var crashlytics: FirebaseCrashlytics
+    private lateinit var customKeySamples: CustomKeySamples
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         val binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        val customKeySamples = CustomKeySamples(applicationContext)
+        this.customKeySamples = CustomKeySamples(applicationContext)
```
Comment (rosariopfernandes) [inline crash/app/src/main/java/com/google/samples/quickstart/crash/kotlin/MainActivity.kt:32]: I believe `this` can be ommited?
```suggestion
        customKeySamples = CustomKeySamples(applicationContext)
```

File: crash/app/src/main/java/com/google/samples/quickstart/crash/java/CustomKeySamples.java [hunk 2]
```diff
@@ -53,18 +65,38 @@ public static void updateAndTrackNetworkState(Context context) {
...
+                connectivityManager.unregisterNetworkCallback(callback);
+                callback = null;
+            }
+        }
+    }
+
+    private void updateNetworkCapabilityCustomKeys(NetworkCapabilities networkCapabilities) {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
             FirebaseCrashlytics instance = FirebaseCrashlytics.getInstance();
             instance.setCustomKey("Network Bandwidth", networkCapabilities.getLinkDownstreamBandwidthKbps());
```
Comment (rosariopfernandes) [inline crash/app/src/main/java/com/google/samples/quickstart/crash/java/CustomKeySamples.java:102]: Maybe we could use `setCustomKeys()` here?

Comment (jakeouellette) [reply crash/app/src/main/java/com/google/samples/quickstart/crash/java/CustomKeySamples.java]: Great idea!

File: crash/app/src/main/java/com/google/samples/quickstart/crash/kotlin/MainActivity.kt (hunk 1 above)
Comment (jakeouellette) [reply crash/app/src/main/java/com/google/samples/quickstart/crash/kotlin/MainActivity.kt]: Definitely.

**Review by rosariopfernandes — APPROVED** [review]

**Comment by jakeouellette** [comment]
Thanks for the feedback, folks!

**Pull request merged**